### PR TITLE
Make URLs secure by default and add parameter to make them unsecure

### PR DESCRIPTION
**Pull request by vvasiloi**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | replaces https://github.com/Sylius/Sylius/pull/11922
| License         | MIT

### CODE REVIEW

**Review by lchrusciel — APPROVED** [review]

File: src/Sylius/Bundle/CoreBundle/Tests/Twig/ChannelUrlExtensionTest.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+use Sylius\Bundle\CoreBundle\Twig\ChannelUrlExtension;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Channel\Model\Channel;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\UrlHelper;
+
+class ChannelUrlExtensionTest extends TestCase
+{
+    public function testGenerateChannelUrlWithChannelHostname(): void
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Tests/Twig/ChannelUrlExtensionTest.php:26]: From what I remember, we are using phpspec notation for PHPUnit tests as well
```suggestion
    /** 
     * @test 
     */
    public function it_generates_channel_url_with_channel_hostname(): void
```

Comment (vvasiloi) [reply src/Sylius/Bundle/CoreBundle/Tests/Twig/ChannelUrlExtensionTest.php:26]: True. I think I copied it from somewhere else...

**Comment by vvasiloi** [comment]
I applied the changes! Thanks for your vigilance. Perhaps we should add some CS rules for things like that.

**Comment by lchrusciel** [comment]
Absolutely we should. Same with `abortIf` in migration. Thanks 👍

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Perhaps, we should add note about this env variable in upgrade file & our documentation

**Comment by lchrusciel** [comment]
Thank you, Victor! :1st_place_medal:

**Pull request merged**